### ISSUE TITLE
error: path '/nix/store/7a47nbg3myjqfl6r89ifr6yz3y04nk25-libidn2-2.3.4-bin' is not valid

### ISSUE THREAD

**Issue by tfmoraes** [NixOS/nixpkgs#257888]
### Steps To Reproduce
1. I'm using flakes with this flake.nix https://github.com/tfmoraes/nixoscfg
2. nixos-rebuild --flake '.#' --use-remote-sudo switch

### Expected behavior
Build new configuration and switch.

### Additional context

```
❯ nixos-rebuild --flake '.#' --use-remote-sudo switch --show-trace 
warning: Git tree '/home/thiago/Sources/nixoscfg' is dirty
building the system configuration...
warning: Git tree '/home/thiago/Sources/nixoscfg' is dirty
error: path '/nix/store/7a47nbg3myjqfl6r89ifr6yz3y04nk25-libidn2-2.3.4-bin' is not valid
```

That are the last lines when calling nixos-rebuild with verbose:

```
copying path '/nix/store/81na6xsrlw9phwhhrhvcznr9145yr722-security-wrapper' from 'https://cache.nixos.org'...
copying path '/nix/store/j780dhssj5crmrcwwfprkwab8jf81y5i-security-wrapper' from 'https://cache.nixos.org'...
building '/nix/store/2bzild6c1sshvkplhf94rnzrcp4c75m5-mounts.sh.drv'...
building '/nix/store/ykfknhxyk9z4h58q85z40d6z190n9rg1-users-groups.json.drv'...
copying path '/nix/store/h4g5xlpk0f7z2brdbj2pdlxsw204y330-exfat-1.4.0' from 'https://cache.nixos.org'...
copying path '/nix/store/dy49ggl1wfcxs6vj75n9xh0x8zn9b21z-lvm2-2.03.22' from 'https://cache.nixos.org'...
building '/nix/store/pvbyf17zrrasf48wncdkgbfjd3pplbpv-append-initrd-secrets.drv'...
copying path '/nix/store/sgzivprbkzkf80p9cqvpzmzlx1pn3w33-groff-1.23.0' from 'https://cache.nixos.org'...
copying path '/nix/store/5q06fsa8as5rp49ys8z3393vqnlmsmas-icu4c-73.2' from 'https://cache.nixos.org'...
copying path '/nix/store/rplkxf10hnl5437hd7dvs933b3161pxj-iodine-unstable-2019-09-27' from 'https://cache.nixos.org'...
copying path '/nix/store/njas4785hp5h91f83glk0sqgfwk5dk5x-jansson-2.14' from 'https://cache.nixos.org'...
copying path '/nix/store/nxcj4f3l4m7li541x5qg7vr0d5xbcm2f-lcms2-2.15' from 'https://cache.nixos.org'...
copying path '/nix/store/ylfhgjp9nd9hb097cxfg98k899dh9q16-ldacBT-2.0.2.3' from 'https://cache.nixos.org'...
copying path '/nix/store/nmhx24x3b4vzj76p15ysx94cysz6zwkp-libICE-1.1.1' from 'https://cache.nixos.org'...
copying path '/nix/store/39dvsflhp75v6m4pnwwl7vhb5g57npi0-libXau-1.0.11' from 'https://cache.nixos.org'...
copying path '/nix/store/n7rw45lry3cmrpvk4y2qvbzk9dxn0vl1-libXdmcp-1.1.4' from 'https://cache.nixos.org'...
copying path '/nix/store/v2110i2agd10a91wi704wzfgdd97jgr4-libatasmart-0.19' from 'https://cache.nixos.org'...
copying path '/nix/store/98s9ygs6c51sa7msax6411n4fs5hfvqc-libdaemon-0.14' from 'https://cache.nixos.org'...
copying path '/nix/store/zsmbs74wpb8krf7mx8kiab1yljx8nfs5-libedit-20221030-3.1' from 'https://cache.nixos.org'...
building '/nix/store/jm1457hixdv9p5jcs3axb9sf65c8awxb-NetworkManager.conf.drv'...
building '/nix/store/gb21nilgihhpyz5w43rxaxmb2h4vnv2f-Xresources-Xft.drv'...
building '/nix/store/jdrc1yvs32mgpdb0cb16za708npfml3b-builder.pl.drv'...
building '/nix/store/37p43ibdiq0zf0vxvzi9m4x2giv0q29i-chfn.pam.drv'...
building '/nix/store/1d4ay9iivra5blnrlnqbn9b37jig6hl6-chpasswd.pam.drv'...
building '/nix/store/6f661c1hdgz2w971wrgqsx7nxmwran96-chsh.pam.drv'...
building '/nix/store/l90f69jqnqq9agm7l3xks21bc29qs6if-client.conf.drv'...
building '/nix/store/w7lczyb8n09cg57nb8djp6d57r25vhav-cups.pam.drv'...
building '/nix/store/n20cgixal9cda0zi9bmf7fsi160r62fg-custom.conf.drv'...
building '/nix/store/dnjmz72ys1n58h4q1ax4kmzkfmwgsyn3-etc-UPower-UPower.conf.drv'...
building '/nix/store/m69fs80c1kaj09xrikqngavxg33q1ngd-etc-X11-xorg.conf.d-00-keyboard.conf.drv'...
building '/nix/store/wbh6k3shiw110pf61g7ninrwpk1wqp2a-etc-fstab.drv'...
building '/nix/store/l0rkhjzvj8jfl5ji8npibk5l84s76i5r-etc-fuse.conf.drv'...
error: path '/nix/store/7a47nbg3myjqfl6r89ifr6yz3y04nk25-libidn2-2.3.4-bin' is not valid
```

### Notify maintainers

@fpletz @vcunat 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.5, NixOS, 23.11 (Tapir), 23.11.20230925.6500b45`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.0`
 - nixpkgs: `/etc/nixpkgs`
```

**Comment by NotAShelf** [comment]
able to reproduce on my own system.

**Comment by CobaltCause** [comment]
This is https://github.com/NixOS/nix/issues/9052

**Comment by Artturin** [comment]
Note that only nixUnstable has been bumped to 2.18

**Comment by tfmoraes** [comment]
Using this tip fixed this problem to me https://github.com/NixOS/nix/issues/9052#issuecomment-1740111801

**Comment by dezren39** [comment]
> Using this tip fixed this problem to me [NixOS/nix#9052 (comment)](https://github.com/NixOS/nix/issues/9052#issuecomment-1740111801)

yes, this issue was worked around for me with the following:
```
nix copy --from https://cache.nixos.org '/nix/store/7a47nbg3myjqfl6r89ifr6yz3y04nk25-libidn2-2.3.4-bin'
nix copy --from https://cache.nixos.org '/nix/store/j7rz908lznabmbk751h7f64679vg8xyp-libidn2-2.3.4-dev'
```

sharing for those looking for a quick solution that land here.
ymmv depending on your exact issues.
